scalding-core/src/main/scala/com/twitter/scalding/Execution.scala

```diff
@@ -531,10 +531,16 @@ object Execution {
    */
   private case class FlowDefExecution(result: (Config, Mode) => FlowDef) extends Execution[Unit] {
     protected def runStats(conf: Config, mode: Mode, cache: EvalCache)(implicit cec: ConcurrentExecutionContext) = {
-      lazy val future = cache
-        .runner
-        .validateAndRun(conf, mode)(result)
-        .map { m => ((), m) }
+      lazy val future = {
+        cache.writer match {
+          case ar: AsyncFlowDefRunner =>
+            ar.validateAndRun(conf, mode)(result).map { m => ((), Map(m)) }
+          case other =>
```

Comment:
We do have a few `#fromFn` uses in our code, specifically in a couple places where we have `FlowDef => Unit` functions to accommodate both `Job` & `Execution`. Deprecation is definitely a good idea though.